src/messaging/ExchangeContext.cpp

```diff
@@ -92,6 +93,32 @@ CHIP_ERROR ExchangeContext::SendMessage(uint16_t protocolId, uint8_t msgType, Pa
     // Set the message type for this header.
     payloadHeader.SetMessageType(msgType);
 
+    payloadHeader.SetInitiator(IsInitiator());
+
+    // If sending via UDP and the auto-request ACK feature is enabled, automatically
+    // request an acknowledgment, UNLESS the NoAutoRequestAck send flag has been specified.
+    if (mReliableMessageContext.AutoRequestAck() && !sendFlags.Has(SendMessageFlags::kSendFlag_NoAutoRequestAck))
+    {
+        payloadHeader.SetNeedsAck(true);
+    }
+
+    // Abort early if Throttle is already set.
+    VerifyOrExit(!mReliableMessageContext.IsThrottling(), err = CHIP_ERROR_SEND_THROTTLED);
```

Comment:
CRMP Throttle Flow mechanism is dropped in the final CRMP spec recently merged. 

This PR is to adapt and integrate current ReliableMessageManager/ReliableMessageContext (which is implemented against original CRMP proposal) to CHIP. https://github.com/project-chip/connectedhomeip/issues/4202 is created to update the CRMP implementation to align with the final merged version.

Reply:
I see.  Asking reviewers to spend a bunch of time to review code that we then plan to immediately remove is really not a good use of their time.  In the interests of conserving reviewer time (which is a _very_ scarce resource at the moment), please roll the core-removal parts of #4202 into this PR, so we are not wasting time on pointless review of that code and are not landing unreviewed code.

Reply:
This is because this PR is posted before the CRMP spec is finalized, I have cleaned up this PR to not adding new stuff which plan to be removed, and https://github.com/project-chip/connectedhomeip/issues/4202 for legacy clean up and new spec update if any.